Sources/GRPCHTTP2Core/Server/GRPCServerStreamHandler.swift

```diff
@@ -163,6 +175,32 @@ extension GRPCServerStreamHandler {
       )
     )
   }
+
+  func channelInactive(context: ChannelHandlerContext) {
+    switch self.stateMachine.unexpectedInboundClose(reason: .channelInactive) {
+    case .forwardStatus:
+      assertionFailure(
+        "`forwardStatus` should only happen on the client side, never on the server."
+      )
+    case .doNothing:
+      context.fireChannelInactive()
+    case .fireError(let error):
+      context.fireErrorCaught(error)
+    }
+  }
+
+  func errorCaught(context: ChannelHandlerContext, error: any Error) {
+    switch self.stateMachine.unexpectedInboundClose(reason: .errorThrown(error)) {
+    case .forwardStatus:
+      assertionFailure(
+        "`forwardStatus` should only happen on the client side, never on the server."
+      )
+    case .doNothing:
+      context.fireErrorCaught(error)
```

Comment:
Hmm, is `doNothing` the right name here? It's a bit confusing that do nothing fires an error here.